Add gitcliff and reuse

Al-an-21 opened the pull request.

Al-an-21 committed:
Added 2 new release tools: git-cliff and reuse

Signed-off-by: Alan Ostrowski <alan.ostrowski@3mdeb.com>

macpijan commented:
@Al-an-21 Please take a look at your MR before submitting for review. It contains much more commmits than just your changes, and in general cannot be reviewed in a sane way. It is likely that you wanted to target `develop` branch - adjusted that.

macpijan commented on README.md:
```diff
@@ -1,3 +1,9 @@
+<!--
+SPDX-FileCopyrightText: 2024 3mdeb <contact@3mdeb.com>
+
+SPDX-License-Identifier: MIT
```
Where does the MIT came from @Al-an-21 ? We clearly already ship Apache license in this repo: https://github.com/Dasharo/open-source-firmware-validation/blob/develop/LICENSE

macpijan reviewed (changes requested):
I have a really difficult time finding git cliff changes here. If there are any, please split them into individual commits from `reuse` changes, so it can be spotted and reviewed separately from auto changes to SPDX headers.

On top of that, if there are some I would appreciate:
- some comments explaining which `ruese` commands have been used to modify the source tree
- some comments explaining whether we needed to do some changes manually, on top of the `reuse` changes

Al-an-21 committed:
Added gitcliff template, updated README, and added 2 tools to requirements files

Signed-off-by: Alan Ostrowski <alan.ostrowski@3mdeb.com>

Al-an-21 committed:
Add license headers for compliance with reuse tool

Signed-off-by: Alan Ostrowski <alan.ostrowski@3mdeb.com>

Al-an-21 commented:
I’ve organized the files and split them across two commits for clarity. Additionally, I updated the license from MIT to Apache 2.0 and expanded the README.md with detailed instructions on using reuse commands for license compliance.

macpijan commented:
> I’ve organized the files and split them across two commits for clarity. Additionally, I updated the license from MIT to Apache 2.0 and expanded the README.md with detailed instructions on using reuse commands for license compliance.

I fail to notice that.

reuse and cliff tools are still mixed up in the same commits, which was my main concern

![image](https://github.com/user-attachments/assets/231db36d-d0dc-44a4-8431-11a51ae6a3ba)

macpijan commented:
@Al-an-21 A friendly reminder that it has not been fix still :disappointed:

DaniilKl commented:
Continuation here https://github.com/Dasharo/open-source-firmware-validation/pull/633.
